src/vs/workbench/services/language/common/languageService.ts

```diff
@@ -178,10 +178,11 @@ class LanguageTableRenderer extends Disposable implements IExtensionFeatureTable
 			localize('grammar', "Grammar"),
 			localize('snippets', "Snippets")
 		];
-		const rows: IRowData[][] = languages.sort((a, b) => a.id.localeCompare(b.id))
+		const rows: IRowData[][] = languages.sort((a, b) => (a.id || '').localeCompare(b.id || ''))
```

Comment:
If `a.id` can be undefined, why does typescript allow to call `localeCompare` on it?
Are you sure this is the root cause of the bug?

Reply:
Yes, because the data is read from package.json which can have `id` undefined.